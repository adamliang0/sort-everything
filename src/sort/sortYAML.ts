import {dump, load} from 'js-yaml';
import type {SortFunction} from './SortFunction';

export const sortYAML: SortFunction = function (text: string) {
  try {
    return {
      payload: dump(load(text), {sortKeys: true}),
      type: 'success',
    };
  } catch (error) {
    return {
      error: error as Error,
      type: 'error',
    };
  }
};
