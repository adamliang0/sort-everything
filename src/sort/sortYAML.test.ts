import { assert, describe, expect, it } from 'vitest';
import { sortYAML } from './sortYAML';

describe('sortYAML', () => {
  it('preserves comments when sorting YAML input', () => {
    const input = `
    variables:
      C_VARIABLE: 'true' # Comment C
      A_VARIABLE: 'true' # Comment A
      B_VARIABLE: 'true' # Comment B
  `;

    const actual = sortYAML(input);
    assert(actual.type === 'success');
    expect(actual.payload).toMatchSnapshot();
  });
});
