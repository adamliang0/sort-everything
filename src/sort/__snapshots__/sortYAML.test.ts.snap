// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sortYAML > preserves comments when sorting YAML input 1`] = `
"variables:
  A_VARIABLE: 'true' # Comment A
  B_VARIABLE: 'true' # Comment B
  C_VARIABLE: 'true' # Comment C
"
`;

exports[`sortYAML preserves comments when sorting YAML input 1`] = `
"variables:
  A_VARIABLE: 'true' # Comment A
  B_VARIABLE: 'true' # Comment B
  C_VARIABLE: 'true' # Comment C
"
`;
